### PR TITLE
Adding content on proxy-based auth

### PR DESCRIPTION
**Pull request by aetter**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jimmyjones2** [comment]
I last used this in pre-1.0 days, but I also needed
```yaml
opendistro_security.auth.type: proxy
```
in kibana.yml

File: docs/security-configuration/proxy.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+      xff:
+        enabled: true
+        internalProxies: '192\.168\.0\.10|192\.168\.0\.11'
+        remoteIpHeader: 'x-forwarded-for'
+```
+
+You can configure the following settings:
+
+Name | Description
+: ---| :---
```
Comment (allenyin96) [inline docs/security-configuration/proxy.md]: I think there's an extra space in ": ---"

File: docs/security-configuration/proxy.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+      xff:
+        enabled: true
+        remoteIpHeader: "x-forwarded-for"
+        internalProxies: '<kibana-ip-address>'
+```
+
+To pass the user and role headers that the authenticating proxy adds from Kibana to the Security plugin, add them to the HTTP header whitelist in `kibana.yml`:
+
+```yml
+elasticsearch.requestHeadersWhitelist: ["securitytenant","Authorization","x-proxy-user","x-proxy-roles"]
```
Comment (jimmyjones2) [inline docs/security-configuration/proxy.md]: I think you need x-forwarded-for here. Also remove Authorization as users should only be using proxy auth?

Comment (aetter) [reply docs/security-configuration/proxy.md]: xff, yep, definitely. Thanks, updated.

My understanding is that you must leave `securitytenant` and `Authorization` as-is, though.

File: docs/security-configuration/proxy.md (hunk 1 above)
Comment (aetter) [reply docs/security-configuration/proxy.md]: Thanks a bunch, resolved.

**Pull request merged**